src/opt.rs

```diff
@@ -245,9 +245,10 @@ impl StorageOpt {
     /// Get the store directory of this configuration. If nothing is specified a cache directory is
     /// used if available, otherwise a temporary directory.
     pub fn store_dir(&self) -> PathBuf {
-        match &self.store_dir {
-            Some(dir) => dir.clone(),
-            None => {
+        match (option_env!("TM_CACHE_DIR"), &self.store_dir) {
```

Comment:
Assuming we want this, I'd rather rename this variable in `TM_STORE_DIR` or something similar. The cache is not the only thing inside there (and it's called `store_dir` in the code).